merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/SimulationDriver.java

```diff
@@ -43,10 +37,16 @@ SimulationResults simulate(
       }
 
       // Schedule the control task.
-      final var controlTask = new ControlTask(schedule);
-      {
-        final var control = engine.initiateTask(elapsedTime, controlTask);
-        engine.scheduleTask(control, elapsedTime);
+      final var taskToPlannedDirective = new HashMap<String, ActivityInstanceId>();
+      for (final var entry : schedule.entrySet()) {
+        final var directiveId = entry.getKey();
+        final var startOffset = entry.getValue().getLeft();
+        final var directive = entry.getValue().getRight();
+
+        final var taskId = engine.initiateTaskFromInput(missionModel, directive);
```

Comment:
I'm curious about the distinction between `initiate` and `instantiate` (and perhaps `initialize`). Care to elaborate? In particular, `initiateTaskFromInput` calls `instantiateDirective`, which makes me wonder if those might mean the same thing.

My own musing: I suppose "initiating" a Task is the domain operation, while "instantiating" a directive quite literally means constructing the Java object representing this directive... but then following that train of thought, I'm curious what the difference between "Start" and "Initiate" would be for a task (since `initiateTaskFromInput` puts a task in the `NotStarted` state - so it has been "initiated", but not "started" 🤔 )

Reply:
Yes, you're poking at a wart that's been bothering me for some time. If you can wait for my next PR, your questions here should be either answered or obviated 🙏 I'm trying to extract the responsibility for instantiating tasks from the simulation engine altogether, so that the Engine can be much more tightly focused on the *scheduling and execution* of tasks. Where a task comes from shouldn't matter to it.